extensions/csharp-v2/operation/method.ts

```diff
@@ -175,13 +175,13 @@ export class OperationMethod extends Method {
 
       yield `// construct URL`;
       urlV = new LocalVariable('_url', dotnet.Var, {
-        initializer: System.Uri.new(`(
+        initializer: System.Uri.new(`${System.Text.RegularExpressions.Regex.declaration}.Replace(
         "${url}"
         ${queryParams.length > 0 ? '+ "?"' : ''}${queryParams.joinWith(pp => `
         + ${removeEncoding(pp, pp.param.name, KnownMediaType.QueryParameter)}`, `
         + "&"`
```

Comment:
Why not use [Array.Join('&')](https://www.tutorialspoint.com/typescript/typescript_array_join)?

Reply:
the  `[EOL]+ "&"` is _actual c# code_ that is getting emitted

the `queryParams` is the (TypeScript) array of query parameters that we have in our hands. 

it's literally using `joinWith` (which is a variant of join that takes a predicate) ..